### ISSUE TITLE
DNS failing from inside container.  Curious to understand what it is I am missing here.  

### ISSUE THREAD

**Issue by hesco** [weaveworks/weave#2360]
This is the code I'm using to run a container:

```
/usr/bin/docker run -d $WEAVE_IP_ASSIGNMENT --name="$DOMAIN" -h $DOMAIN $DOCKER_RUN_OPTS $PORTS $ATTACH_VOLUMES $IMAGE
```

Its getting evaluated like so in one representative sample:

```
Successfully built fa493d99fe3e

+ /usr/bin/docker run -d --ip=10.0.1.31 --net weave --dns 172.17.0.1 --dns-search=weave.local. --name=smtp.example.com -h smtp.example.com 
--memory=1gb --restart=always --privileged=true -p 199.101.183.162:25:25/tcp -v /data/etc/postfix/smtp.example.com:/etc/postfix -v /data/var/log/smtp.example.com/mail.log:/var/log/mail.log smtp:latest

2e72b78b81708f737c614c1f57ce7958d038c7e45da2f1378e4d1cecfbdc4869

+ /root/docker-helper-scripts/lib/sh/ip.sh -c smtp.example.com -a
docker: --.--.--.--
ipam: <no value>
weave: 10.0.1.31
+ exit 0

```

and I see this inside the container:

```
root@smtp:~# ping gmail.com

ping: unknown host gmail.com

root@smtp:~# cat /etc/resolv.conf                                                                                                                                       

search weave.local
nameserver 127.0.0.11
options ndots:0

root@smtp:~# ip route show 

default via 172.18.0.1 dev eth1 
10.0.0.0/20 dev ethwe0  proto kernel  scope link  src 10.0.1.31 
172.18.0.0/16 dev eth1  proto kernel  scope link  src 172.18.0.5 
224.0.0.0/4 dev ethwe0  scope link
```

**Comment by awh** [comment]
Thanks for the report @hesco.

I have been able to reproduce this on Ubuntu 16.04 when the router is restarted automatically by Docker after a reboot - the issue is that at the time the router is started, `/etc/resolv.conf` (which is a symlink to `/run/resolvconf/resolv.conf`) hasn't been populated yet, so the router finds no fallback DNS servers to use. Immediately after such a reboot:
```
$ weave report -f '{{.DNS.Upstream}}'
[]
```
If I launch containers with `--net weave` I cannot resolve e.g. gmail.com as in your example. However, if I then manually `weave stop-router && weave launch-router` I get this:
```
$ sudo weave report -f '{{.DNS.Upstream}}'
[127.0.1.1]
```
after which I can resolve non-weave.local addresses correctly.

Does this match what you're seeing?

**Comment by rade** [comment]
@hesco ping

**Comment by rade** [comment]
I am going to assume that this is fixed by #2364.

**Comment by hesco** [comment]
Perhaps this has been fixed, but in working through the steps laid out by @awh , I get before any router resets:

```
root@dessalines011:~# weave stop-router && weave launch-router 
911e11f44cc84e0ffffc227b0c74130ddb647dca3259a078f5f449d9ed7fdc5d
root@dessalines011:~# weave report -f '{{.DNS.Upstream}}' 
[68.64.241.98 68.64.241.102]root@dessalines011:~# 
root@dessalines011:~#```

but
